core/state/state_object.go

```diff
@@ -65,9 +55,15 @@ type stateObject struct {
 	trie Trie   // storage trie, which becomes non-nil on first access
 	code []byte // contract bytecode, which gets set when code is loaded
 
-	originStorage  Storage // Storage cache of original entries to dedup rewrites
-	pendingStorage Storage // Storage entries that need to be flushed to disk, at the end of an entire block
-	dirtyStorage   Storage // Storage entries that have been modified in the current transaction execution, reset for every transaction
+	originStorage  Storage // Storage entries that have been accessed within the current block
+	dirtyStorage   Storage // Storage entries that have been modified within the current transaction
+	pendingStorage Storage // Storage entries that have been modified within the current block
+
+	// needCommit tracks a set of storage entries that have been modified but
+	// not yet committed, along with their original values before mutation.
+	//
+	// Note, this map will be reset for each transaction before byzantium fork.
+	needCommit Storage
```

Comment:
Modified but not yet committed, this sounds a lot like `dirtyStorage`. Please add some clarification so a reader understands what's different between those two things